### PR TITLE
Set snippet to actual line

### LINKED ISSUE #4: Algorithm misidentification
Hi, I'm using an `output.sarif` file with both md5 and sha1 results in it.
```
    "results" : [ {
      "ruleId" : "py/quantum-readiness/cbom/all-cryptographic-algorithms",
      "ruleIndex" : 1,
      "rule" : {
        "id" : "py/quantum-readiness/cbom/all-cryptographic-algorithms",
        "index" : 1
      },
      "message" : {
        "text" : "Use of algorithm MD5"
      },
      "locations" : [ {
        "physicalLocation" : {
          "artifactLocation" : {
            "uri" : "sqli/dao/dummy.py",
            "uriBaseId" : "%SRCROOT%",
            "index" : 0
          },
          "region" : {
            "startLine" : 1,
            "startColumn" : 21,
            "endColumn" : 24
          },
          "contextRegion" : {
            "startLine" : 1,
            "endLine" : 3,
            "snippet" : {
              "text" : "from hashlib import md5\r\nfrom hashlib import sha1\r\nfrom typing import NamedTuple, Optional\r\n"
            }
          }
        }
      } ],
      "partialFingerprints" : {
        "primaryLocationLineHash" : "a3859dd903b217e8:1",
        "primaryLocationStartColumnFingerprint" : "20"
      }
    }, {
      "ruleId" : "py/quantum-readiness/cbom/all-cryptographic-algorithms",
      "ruleIndex" : 1,
      "rule" : {
        "id" : "py/quantum-readiness/cbom/all-cryptographic-algorithms",
        "index" : 1
      },
      "message" : {
        "text" : "Use of algorithm SHA1"
      },
      "locations" : [ {
        "physicalLocation" : {
          "artifactLocation" : {
            "uri" : "sqli/dao/dummy.py",
            "uriBaseId" : "%SRCROOT%",
            "index" : 0
          },
          "region" : {
            "startLine" : 2,
            "startColumn" : 21,
            "endColumn" : 25
          },
          "contextRegion" : {
            "startLine" : 1,
            "endLine" : 4,
            "snippet" : {
              "text" : "from hashlib import md5\r\nfrom hashlib import sha1\r\nfrom typing import NamedTuple, Optional\r\n\r\n"
            }
          }
        }
      } ],
```

However I can see that the snippet which is being provided for both is:
```
"snippet" : {
  "text" : "from hashlib import md5\r\nfrom hashlib import sha1\r\nfrom typing import NamedTuple, Optional\r\n\r\n"
```

This means that when `utils.py` performs:
```
def get_algorithm(code_snippet):
    match = _ALGORITHM_REGEX.search(code_snippet)
    if match:
        return match.group()
    return 'unknown'
```

The match is "MD5" on both occasions.

**MD5 was the finding, MD5 detected as algorithm**
image

**SHA1 was the finding, MD5 detected as algorithm**
image

The actual code in the script is:
```
from hashlib import md5
from hashlib import sha1
from typing import NamedTuple, Optional

from aiopg import Connection
...
...
```

I shall look look into whether the snippet can be less verbose,

> [--[no-]sarif-add-snippets](https://docs.github.com/en/code-security/codeql-cli/codeql-cli-manual/database-analyze#--no-sarif-add-snippets)
> 
> [SARIF formats only] Include code snippets for each location mentioned in the results, with two lines of context before and after the reported location.  

(https://docs.github.com/en/code-security/codeql-cli/codeql-cli-manual/database-analyze#--no-sarif-add-snippets)

Because at the moment this can cause a problem for the `match = _ALGORITHM_REGEX.search(code_snippet)` because `re.search` just finds the first match (vs `.findall` which will return a list of all non-overlapping matches in the input string).

So with this particular code, when the cbom and cryptocheck are generated, I'm only ever getting the MD5 finding.

### PR DESCRIPTION
**Pull request by mtcolman**
Code to ensure snippet accurately reflects the identified line of code. Resolves #4 .

### CODE REVIEW

**Comment by mtcolman** [comment]
@emilejq thanks for the feedback. I believe all requested/suggested changes have been implemented.

File: cbom/parser/algorithm.py [hunk 1]
```diff
@@ -44,8 +44,9 @@ def parse_algorithm(cbom, codeql_result):
 
 
 def _generate_crypto_component(codeql_result):
+    algorithm = utils.get_algorithm(utils.extract_precise_snippet(codeql_result['locations'][0]['physicalLocation']['contextRegion']['snippet']['text'], codeql_result['locations'][0]['physicalLocation']['region']))
```
Comment (emilejq) [inline cbom/parser/algorithm.py:47]:
```python
    code_snippet = codeql_result['locations'][0]['physicalLocation']['contextRegion']['snippet']['text']
    algorithm = utils.get_algorithm(utils.extract_precise_snippet(code_snippet, codeql_result['locations'][0]['physicalLocation']['region']))
```

File: cbom/parser/utils.py [hunk 2]
```diff
@@ -47,3 +47,25 @@ def merge_code_snippets(dc1, dc2):
...
+        case 1: array_of_lines = snippet.split('\n')
+        case 2: array_of_lines = snippet.split('\n')[1:]
+        case _: array_of_lines = snippet.split('\n')[2:]
+
+    if not line_end:
+        actual_line = array_of_lines[0]
+        return actual_line[line_start_col - 1:line_end_col]
+    else:
+        end_line_index = (line_end - line_start)
+        actual_lines = array_of_lines[0:end_line_index + 1]
```
Comment (emilejq) [inline cbom/parser/utils.py:68]: `actual_lines = array_of_lines[:end_line_index + 1]`

**Comment by emilejq** [comment]
Thanks! Just resolve those final two comments please and this is then good to be merged

**Comment by mtcolman** [comment]
@emilejq think it's done now 👍

**Review by emilejq — APPROVED** [review]

**Pull request merged**